python/tests/test_snippets.py

```diff
@@ -54,6 +54,50 @@ def test_floor_snippet(house: etree.ElementTree) -> None:
     }
 
 
+def test_wall_snippet(house: etree.ElementTree) -> None:
+    output = snippets.snip_house(house)
+    assert 'walls' in output
+    assert len(output['walls']) == 3
+    assert sorted(output['walls'], key=lambda row: row['label'])[0] == {
+        'label': 'End Wall',
+        'constructionTypeCode': 'Code 1',
+        'constructionTypeValue': '1201101121',
+        'effectiveRsi': '1.7435',
+        'nominalRsi': '1.432',
+        'perimeter': '7.367',
+        'height': '1.2283',
+    }
+
+
+def test_user_specified_wall_snippet() -> None:
+    xml_text = """
+<House><Components>
+    <Wall>
+        <Label>Test Floor</Label>
+        <Construction>
+            <Type rValue="2.6892" nominalInsulation="3.3615">User specified</Type>
```

Comment:
Note the lack of an `idref` attribute here. This was unexpected, but is because this particular feature was a "user specified" wall code.

Reply:
I notice that this wall also lacks a `Measurements` node, and so has no `height` and `perimeter`, does that go along with a "user specified" wall code, or is this just testing that the snippet extractor can handle the missing node?

Reply:
Kind of an extra benefit. The main purpose of the test is to check the case where there is no `idref` attribute because the value is `User specified`.